crates/pallet-domains/src/domain_registry.rs

```diff
@@ -0,0 +1,169 @@
+use frame_support::weights::Weight;
+use frame_support::{ensure, PalletError};
+use scale_info::TypeInfo;
+use sp_core::Get;
+use sp_domains::{DomainId, GenesisDomain, RuntimeId};
+use sp_runtime::traits::{CheckedAdd, Zero};
+use sp_std::vec::Vec;
+
+const DOMAIN_INSTANCE_ID: LockIdentifier = *b"instance";
+
+pub type EpochIndex = u32;
+
+/// Domain registry specific errors
+#[derive(TypeInfo, Encode, Decode, PalletError, Debug, PartialEq)]
+pub enum Error {
+    InvalidBundlesPerBlock,
+    ExceedMaxBlockWeight,
+    ExceedMaxBlockSize,
+    MaxDomainId,
+    InvalidSlotProbability,
+    RuntimeNotFound,
+    InsufficientFund,
+}
+
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct DomainConfig {
+    // TODO: should we limit how long the domain name can be?
+    /// A user defined name for this domain, should be a human-readable UTF-8 encoded string.
+    pub domain_name: Vec<u8>,
+    /// A pointer to the `RuntimeRegistry` entry for this domain.
+    pub runtime_id: RuntimeId,
+    /// The max block size for this domain, may not exceed the system-wide `MaxBlockSize` limit.
+    pub max_block_size: u32,
+    /// The max block weight for this domain, may not exceed the system-wide `MaxBlockWeight` limit.
+    pub max_block_weight: Weight,
+    /// The probability of successful bundle in a slot (active slots coefficient). This defines the
+    /// expected bundle production rate, must be `> 0` and `≤ 1`.
+    pub bundle_slot_probability: (u64, u64),
+    /// The expected number of bundles for a domain block, must be `≥ 1` and `≤ MaxBundlesPerBlock`.
+    pub target_bundles_per_block: u32,
+}
+
+impl DomainConfig {
+    pub(crate) fn from_genesis<T: Config>(
+        genesis_domain: &GenesisDomain<T::AccountId>,
+        runtime_id: RuntimeId,
+    ) -> Self {
+        DomainConfig {
+            domain_name: genesis_domain.domain_name.clone(),
+            runtime_id,
+            max_block_size: genesis_domain.max_block_size,
+            max_block_weight: genesis_domain.max_block_weight,
+            bundle_slot_probability: genesis_domain.bundle_slot_probability,
+            target_bundles_per_block: genesis_domain.target_bundles_per_block,
+        }
+    }
+}
+
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct DomainInstance<Number, Hash, AccountId> {
+    /// The address of the domain creator, used to validate updating the domain config.
+    pub owner_account_id: AccountId,
+    /// The consensus chain block number when the domain first instantiated.
+    pub created_at: Number,
+    /// The hash of the genesis block for this domain.
+    pub genesis_block_hash: Hash,
+    /// The domain genesis config.
+    pub domain_config: DomainConfig,
+}
+
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct StakeSummary<Balance, OperatorId> {
+    /// Index of the current epoch for the operator election for this domain.
```

Comment:
Leave todo here as this will change in my Staking work. Would be great if we can avoid staking things here and leave some todos to fix later so that we dont have duplicates